ratatool-scalacheck/src/test/scala/com/spotify/ratatool/scalacheck/AvroGeneratorTest.scala

```diff
@@ -95,18 +103,30 @@ class AvroGeneratorTest extends AnyFlatSpec with Matchers with ScalaCheckPropert
       .name("javaStringField").`type`().stringBuilder().prop(GenericData.STRING_PROP, "String").endString().noDefault()
       .name("charSequenceField").`type`().stringBuilder().prop(GenericData.STRING_PROP, "CharSequence").endString().noDefault()
       .name("utf8Field").`type`().stringBuilder().prop(GenericData.STRING_PROP, "CharSequence").endString().noDefault()
+      .name("stringableBigDecimalField").`type`().stringBuilder().prop(SpecificData.CLASS_PROP, classOf[java.math.BigDecimal].getName).endString().noDefault()
+      .name("stringableDefaultField").`type`().stringBuilder().prop(SpecificData.CLASS_PROP, classOf[StringableCustom].getName).endString().noDefault()
       .name("mapDefaultKeyField").`type`().map().values().longType().noDefault()
       .name("mapJavaStringKeyField").`type`().map().prop(GenericData.STRING_PROP, "String").values().longType().noDefault()
+      .name("mapStringableBigIntegerKeyField").`type`().map().prop(SpecificData.KEY_CLASS_PROP, classOf[java.math.BigInteger].getName).values().longType().noDefault()
+      .name("mapStringableDefaultKeyField").`type`().map().prop(SpecificData.KEY_CLASS_PROP, classOf[StringableCustom].getName).values().longType().noDefault()
       .endRecord()
     // format: on
 
-    val gen = avroOf(schema)
+    val stringableGens = Map[Class[_], Gen[_]](
+      classOf[java.math.BigDecimal] -> Gen.choose(BigDecimal(-1), BigDecimal(1)).map(_.bigDecimal),
+      classOf[java.math.BigInteger] -> Gen
+        .choose(BigInt(0), BigInt(Long.MaxValue))
+        .map(_.bigInteger)
+    )
+    val gen = avroOf(schema, stringableGens)
```

Comment:
without those custom `stringableGens`, random strings would be generated to create `BigDecimal` and `BigInteger` that most likely creates a parse exception.
Here we give custom generators for those types and bypass the conversion from string.

The `StringableCustom` type not being part of the map will be constructed with random string